chain-impl-mockchain/src/leadership/bft.rs

```diff
@@ -0,0 +1,147 @@
+use crate::block::SignedBlock;
+use crate::key::PublicKey;
+use crate::leadership::IsLeading;
+use crate::update::{BftSelectionDiff, ValueDiff};
+
+use chain_core::property::{self, LeaderSelection};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BftRoundRobinIndex(usize);
+
+/// The BFT Leader selection is based on a round robin of the expected leaders
+#[derive(Debug)]
+pub struct BftLeaderSelection<LeaderId> {
+    my: Option<BftRoundRobinIndex>,
+    leaders: Vec<LeaderId>,
+
+    current_leader: LeaderId,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    BlockHasInvalidLeader(PublicKey, PublicKey),
```

Comment:
It would be more informative to have a structure with named fields, but this will do for the mockchain.